sdk/storage/Azure.Storage.Common/tests/Shared/StorageRecordedTestSanitizer.cs

```diff
@@ -22,100 +23,19 @@ public class StorageRecordedTestSanitizer : RecordedTestSanitizer
-
-            // Remote copy source authorization header
-            if (headers.ContainsKey(CopySourceAuthorization))
-            {
-                headers[CopySourceAuthorization] = new string[] { SanitizeValue };
-            }
-
-            // Santize any copy source
-            if (headers.TryGetValue(CopySourceName, out var copySource))
-            {
-                headers[CopySourceName] = copySource.Select(c => SanitizeUri(c)).ToArray();
-            }
-
-            if (headers.TryGetValue(RenameSource, out var renameSource))
-            {
-                headers[RenameSource] = renameSource.Select(c => SanitizeQueryParameters(c)).ToArray();
-            }
-
-            if (headers.TryGetValue(PreviousSnapshotUrl, out var snapshotUri))
-            {
-                headers[PreviousSnapshotUrl] = snapshotUri.Select(c => SanitizeUri(c)).ToArray();
-            }
-
-            if (headers.TryGetValue(FileRenameSource, out var fileRenameSource))
-            {
-                headers[FileRenameSource] = fileRenameSource.Select(c => SanitizeQueryParameters(c)).ToArray();
-            }
-        }
+            UriRegexSanitizers.Add(UriRegexSanitizer.CreateWithQueryParameter(SignatureQueryName, SanitizeValue));
 
-        public override string SanitizeTextBody(string contentType, string body)
-        {
-            if (contentType.Contains("urlencoded"))
-            {
-                try
-                {
-                    // If it's been URL encoded, make sure it doesn't contain
-                    // a client_secret
-                    var builder = new UriBuilder() { Query = body };
-                    var query = new UriQueryParamsCollection(body);
-                    if (query.ContainsKey("client_secret"))
-                    {
-                        query["client_secret"] = SanitizeValue;
-                    }
-                    return query.ToString();
-                }
-                catch
-                {
-                }
-            }
+            HeaderRegexSanitizers.Add(new HeaderRegexSanitizer("x-ms-encryption-key", SanitizeValue));
+            HeaderRegexSanitizers.Add(new HeaderRegexSanitizer(CopySourceAuthorization, SanitizeValue));
+            HeaderRegexSanitizers.Add(HeaderRegexSanitizer.CreateWithQueryParameter(CopySourceName, SignatureQueryName, SanitizeValue));
+            HeaderRegexSanitizers.Add(HeaderRegexSanitizer.CreateWithQueryParameter(RenameSource, SignatureQueryName, SanitizeValue));
+            HeaderRegexSanitizers.Add(HeaderRegexSanitizer.CreateWithQueryParameter(PreviousSnapshotUrl, SignatureQueryName, SanitizeValue));
+            HeaderRegexSanitizers.Add(HeaderRegexSanitizer.CreateWithQueryParameter(FileRenameSource, SignatureQueryName, SanitizeValue));
 
-            // If anything goes wrong, don't sanitize
-            return body;
-        }
-
-        public override string SanitizeVariable(string variableName, string environmentVariableValue) =>
```

Comment:
This doesn't appear to be necessary as it is handled here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Common/tests/Shared/TenantConfiguration.cs#L113

I re-recorded tests locally to verify that the connection string in the variables is still sanitized.